src/sieve.py

```diff
@@ -16,5 +19,13 @@ def sieve(n: int) -> list[int]:
     primes = []
 
     # FIXME: fill out this bit
-
+    while len(candidates)>0:
+        prime  =candidates[0]
+        del candidates[0]
```

Comment:
Generally, `del x[i]` is a slow operation for lists, so it is best avoided. We haven't seen why, yet, so you will have to take me on my word on this.

You do, however, not have to delete `candidates[0]`. You collect all numbers `i % prime != 0`, but `candidate[0] % candidate[0] == 0` and the value will never go into `not_deleted` anyway.